sdk/src/main/java/com/hedera/hashgraph/sdk/Executable.java

```diff
@@ -289,10 +293,14 @@ void advanceRequest() {
 
     boolean shouldRetryExceptionally(@Nullable Throwable error) {
         if (error instanceof StatusRuntimeException) {
-            var status = ((StatusRuntimeException) error).getStatus().getCode();
-
-            return status.equals(io.grpc.Status.UNAVAILABLE.getCode())
-                || status.equals(io.grpc.Status.RESOURCE_EXHAUSTED.getCode());
+            var statusException = (StatusRuntimeException) error;
+            var status = statusException.getStatus().getCode();
+            var description = statusException.getStatus().getDescription();
+
+            return (status == io.grpc.Status.Code.NOT_FOUND) ||
```

Comment:
We shouldn't be retrying on `NOT_FOUND` for regular Hedera Services requests, only `TopicMessageQuery`